src/Learning/KWDataPreparation/KWClassStats.cpp

```diff
@@ -828,23 +829,26 @@ void KWClassStats::WriteReport(ostream& ost)
 	    << "\t" << TSV::Export(GetClass()->GetName()) << "\n";
 
 	// Nombres d'attributs par type
-	ost << "Variables"
-	    << "\n";
-	nTotalAttributeNumber = 0;
-	for (nType = 0; nType < KWType::None; nType++)
+	if (IsStatsComputed())
 	{
-		if (KWType::IsData(nType))
+		ost << "Variables"
+		    << "\n";
+		nTotalAttributeNumber = 0;
+		for (nType = 0; nType < KWType::None; nType++)
 		{
-			nAttributeNumber = GetUsedAttributeNumberForType(nType);
-			nTotalAttributeNumber += nAttributeNumber;
-			if (nAttributeNumber > 0)
-				ost << "\t" << KWType::ToString(nType) << "\t" << nAttributeNumber << "\n";
+			if (KWType::IsData(nType))
+			{
+				nAttributeNumber = GetUsedAttributeNumberForType(nType);
+				nTotalAttributeNumber += nAttributeNumber;
+				if (nAttributeNumber > 0)
+					ost << "\t" << KWType::ToString(nType) << "\t" << nAttributeNumber << "\n";
+			}
 		}
+		ost << "\t"
+		    << "Total"
+		    << "\t" << nTotalAttributeNumber << "\n";
```

Comment:
I would put these on the same line to be consistent with other places where we write to the output stream (like e.g. line 866).

Reply:
Ce n'est pas possiblle: c'est le pretty printer qui décide de la mise en forme.